src/bp/core/services/cms.ts

```diff
@@ -578,15 +578,49 @@ export class CMSService implements IDisposeOnExit {
 
   // This methods finds the translated property and returns the original properties
   private getOriginalProps(formData: object, contentType: ContentType, lang: string, defaultLang?: string) {
-    const originalProps = Object.keys(_.get(contentType, 'jsonSchema.properties'))
+    const schema = <FormDefinition>contentType.newSchema
+    const fields = [...schema.fields, ...schema.advancedSettings]
+    const result = _.cloneDeep(formData)
 
-    if (originalProps) {
-      return originalProps.reduce((result, key) => {
-        result[key] = formData[key + '$' + lang] || (defaultLang && formData[key + '$' + defaultLang])
-        return result
-      }, {})
-    } else {
-      return formData
```

Comment:
The reason we had to read the schema is because we had no idea which properties needed to be translated. The format was something like
```js
formData: {
  'text$en': 'sometext',
  'typing$en': true,
  'text$fr': 'texte'
}
```

And we don't have to do this anymore with the new structure, which looks like
```js
formData: {
  text: {
    en: 'sometext',
    fr: 'texte'
  }
}
```

The logic is easier: if it's an object, check if object[lang] is set. If not, check if object[defaultLang] is set. If not, then just return object.

I don't think we should migrate old bots to the new structure, too much things to test